iina/PlayerCore.swift

```diff
@@ -374,15 +381,36 @@ class PlayerCore: NSObject {
   }
 
   // Terminate mpv
-  func terminateMPV(sendQuit: Bool = true) {
+  func terminateMPV() {
     guard !isMpvTerminated else { return }
+    isMpvTerminated = true
     savePlaybackPosition()
     invalidateTimer()
```

Comment:
> But what you really want is both of those in one atomic operation (e.g., getAndSet()

YES! You are pointing out one of the reasons why I mentioned I'm worried the "fix" is not deterministic. To make this fully deterministic requires locking similar to the proposed logger fix in PR #3938. I started down that path, but was concerned about all the changes required to do that. In particular, how do you handle "get" operations that occur after mpv has been told to shutdown? If designing from scratch probably you would make all results optional and return nothing. A lot of work to retrofit that.

A different approach can be seen in the replacement PR. That one takes the attitude that this method must only be called by the main thread. So the thread in `MPVController` processing mpv events has to queue calls to `PlayerCore` to the main dispatch queue. Probably some internal checks should be added to ensure use of the main thread.

I too am used to making a class "iron clad". Written from scratch a call to stop after quit would be an internal error. Always painful to try and add such enforcement after the fact. In this case `AppDelegate` is supposed to ensure that does not happen. I am concerned about the flow if a file is being opened right as IINA starts to shutdown.

Initially I was thinking an event driven state machine and ended up with the thread based termination sequence seen in this PR. The new PR is event driven, but does not have a state machine.

At the moment there is no coordination with a high chance a race condition causes a crash. I'm trying to fix that enough so failures are unlikely without all the changes required to fully refactor the code. Basically trying to avoid the massive increase in the scope of changes needed to really nail this down.

For a fully "safe" system we'd have to figure out what to do about mpv deciding on its own to shutdown, as in when the user sends a `quit` command directly to mpv. That opens a window where mpv is shutting down and IINA doesn't know that until mpv sends the shutdown event. IINA might access mpv in that window triggering the same kind of crash we are trying to fix.